packages/backend/src/managers/gitManager.ts

```diff
@@ -33,4 +33,63 @@ export class GitManager {
       await simpleGit(gitCloneInfo.targetDirectory).checkout([gitCloneInfo.ref]);
     }
   }
+
+  async getRepositoryRemotes(directory: string): Promise<RemoteWithRefs[]> {
+    return simpleGit(directory).getRemotes(true);
+  }
+
+  async getRepositoryStatus(directory: string): Promise<StatusResult> {
+    return simpleGit(directory).status();
+  }
+
+  async isRepositoryUpToDate(
+    directory: string,
+    origin: string,
+    ref?: string,
+  ): Promise<{ ok?: boolean; error?: string }> {
+    const remotes: RemoteWithRefs[] = await this.getRepositoryRemotes(directory);
+
+    if (!remotes.some(remote => remote.refs.fetch === origin)) {
+      return {
+        error: `The local repository does not have remote ${origin} configured. Remotes: ${remotes
+          .map(remote => `${remote.name} ${remote.refs.fetch} (fetch)`)
+          .join(',')}`,
+      };
+    }
+
+    const status: StatusResult = await this.getRepositoryStatus(directory);
+
+    if (status.detached) {
+      return { error: 'The local repository is detached.' };
+    }
+
+    if (status.modified.length > 0) {
+      return { error: 'The local repository has modified files.' };
+    }
+
+    if (status.ahead !== 0) {
+      return {
+        error: `The local repository has ${status.ahead} commit(s) ahead.`,
+      };
+    }
+
+    // Ensure the branch tracked is the one we want
+    if (ref !== undefined && status.tracking !== ref) {
```

Comment:
@jeffmaury here I am ensuring that the reference provided by the recipe catalog is matching the one expected

Reply:
As I see no changes, here are the scenarios that I think we should support:

- switch from one commit to another
- switch from one tag to another
- optionaly switch from one branch to a commit or tag (as we don't want to have recipes on branches as it's not stable